qubes_menu/appmenu.py

```diff
@@ -252,13 +252,18 @@ def _key_press(self, _widget, event):
         """
         Keypress handler, to allow closing the menu with an ESC key and to fix
         some issues with space (as we have search by default, we should not
-        react to space with launching an app).
+        react to space with launching an app, but instead put the space
+        into the search bar).
         """
         if event.keyval == Gdk.KEY_Escape:
             self.hide_menu()
         if event.keyval == Gdk.KEY_space:
-            if not isinstance(self.get_active_window().get_focus(),
-                              Gtk.SearchEntry):
+            current_widget = self.get_active_window().get_focus()
+            if isinstance(current_widget,
+                          Gtk.SearchEntry):
+                p = current_widget.get_position()
+                current_widget.insert_text(" ", p)
+                current_widget.set_position(p + 1)
                 return True
```

Comment:
I suppose `current_widget` is the same as the widget receiving the event? If so, this return value (whether to stop propagation) seems backwards: the search widget should continue handling the event (by inserting a space itself), while the menu items should stop handling the event. The code before this change makes more sense to me.